do-not-commit/subprocess-shell-true.py

```diff
@@ -0,0 +1,18 @@
+# cf. https://github.com/returntocorp/semgrep/blob/develop/docs/writing_rules/examples.md#auditing-dangerous-function-use
+import subprocess
+import sys
+
+subprocess.call("echo 'hello'")  # Doesn't match
+
+subprocess.call("grep -R {} .".format(sys.argv[1]))  # Doesn't match
+
+# ruleid:subprocess-shell-true
+subprocess.call("grep -R {} .".format(sys.argv[1]), shell=True)  # Matches here
```

Comment:
This is not checking the return value of this subprocess call; if it fails no exception will be raised. Consider subprocess.check_call() instead ([.unchecked-returns.unchecked-subprocess-call](https://semgrep.dev/r/.unchecked-returns.unchecked-subprocess-call))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=.unchecked-returns.unchecked-subprocess-call) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>